### PR TITLE
Fail on return code

### PR DESCRIPTION
**Pull request by stefanmb**
This PR reintroduces https://github.com/Shopify/doggy/pull/42, while allowing `find` to receive a `404`.

Doggy must absolutely *fail* if it cannot sync correctly, otherwise it will fail to update dashboards/monitors with syntax errors, which can cause severe ops disruptions.

Resolves: https://github.com/Shopify/doggy/issues/43

### CODE REVIEW

**Comment by stefanmb** [comment]
@Shopify/traffic

**Comment by stefanmb** [comment]
It should be pointed out this only partially fixes the problem. Doggy won't even try to update if the `sha` doesn't look updated, so we potentially have many objects that have syntax errors in them.

File: lib/doggy/model.rb [hunk 1]
```diff
@@ -23,7 +23,8 @@ class << self
       attr_accessor :root
 
       def find(id)
-        attributes = request(:get, resource_url(id))
+        attributes = nil
```
Comment (ElvinEfendi) [inline lib/doggy/model.rb]: Why is this needed?

Comment (stefanmb) [inline lib/doggy/model.rb]: It isn't!

**Comment by stefanmb** [comment]
@ElvinEfendi That test is already included in `master`.  (That PR was merged.)

File: lib/doggy/model.rb (hunk 1 above)
Comment (ElvinEfendi) [inline lib/doggy/model.rb]: Can we remove it then?

Comment (stefanmb) [inline lib/doggy/model.rb]: Yes, of course. Working on the test issues.

**Comment by ElvinEfendi** [comment]
>  That test is already included in master. (That PR was merged.)

Ah right, all good then :shipit:

**Comment by stefanmb** [comment]
@charlescng I added some more tests + fixed a bug. @ElvinEfendi Could I have your help publishing the new version? I don't think I have the permissions.

**Comment by ElvinEfendi** [comment]
> @ElvinEfendi Could I have your help publishing the new version? I don't think I have the permissions.

@stefanmb sure, I'll release the new version once you merge it.
